lib/IMAP/Sync/Synchronizer.php

```diff
@@ -134,18 +134,12 @@ private function getChangedMessageUids(Horde_Imap_Client_Base $imapClient, Horde
 
 		// Without QRESYNC we need to specify the known ids and in oder to avoid
 		// overly long IMAP commands they have to be chunked.
-		return array_merge(
-			[], // for php<7.4 https://www.php.net/manual/en/function.array-merge.php
-			...array_map(
-				static function (array $uids) use ($imapClient, $mailbox, $request) {
-					return $imapClient->sync($mailbox, $request->getToken(), [
-						'criteria' => Horde_Imap_Client::SYNC_FLAGSUIDS,
-						'ids' => new Horde_Imap_Client_Ids($uids),
-					])->flagsuids->ids;
-				},
-				array_chunk($request->getUids(), self::UID_CHUNK_SIZE)
-			)
-		);
+		$sorted = $request->getUids();
+		sort($sorted);
+		return $imapClient->sync($mailbox, $request->getToken(), [
+			'criteria' => Horde_Imap_Client::SYNC_FLAGSUIDS,
+			'ids' => new Horde_Imap_Client_Ids(min($sorted) . ':' . max($sorted)),
```

Comment:
the sync might be triggered by the front-end. In that case we only ask for a sync of the messages currently shown, like the 20 newest messages of the mailbox. *Newest* is defined by the message send date, **not** the UID. So the range might fetch messages that are not even relevant.

Example:

0) INBOX has 200 messages (UID 0:199), Archive has 5000 messages (UID 0:4999)
1) User moves archived messages back to INBOX, archive messages get new UIDs 200:4299
2) User receives a new message in INBOX, it gets UID 4300
3) Front-end is open, shows UID 4300, 199, 198 ... 180
4) Front-end requests flag sync of those 20 messages
5) Backend converts range to 180:4300 -> full mailbox is fetched

The question is how often this happens. If someone never moves messages around the UID correlate with the message date in terms of order.

OTOH we can introduce two paths:

1) If more UIDS than self::UID_CHUNK_SIZE are checked for flags we use your range request because this is very likely a full mailbox sync anyway
2) If less than self::UID_CHUNK_SIZE UIDs are checked don't bother to optimize and send the literal UIDs